net/quagga/src/opnsense/service/templates/OPNsense/Quagga/bgpd.conf

```diff
@@ -0,0 +1,36 @@
+{% if helpers.exists('OPNsense.quagga.bgp.enabled') and OPNsense.quagga.bgp.enabled == '1' %}
+!
+! Zebra configuration saved from vty
+!   2017/03/03 20:21:04
+!
+!
+!
+!
+{% if helpers.exists('OPNsense.quagga.bgp.asnumber') and OPNsense.quagga.bgp.asnumber != '' %}
+router bgp {{ OPNsense.quagga.bgp.asnumber }}
+{% if helpers.exists('OPNsense.quagga.bgp.networks') %}
+{%   for network in OPNsense.quagga.bgp.networks.split(',') %}
+ network {{ network }}
+{%   endfor %}
+{%  endif %}
+{% if helpers.exists('OPNsense.quagga.bgp.neighbors.neighbor') %}
+{%   for neighbor in helpers.toList('OPNsense.quagga.bgp.neighbors.neighbor') %}
+{%     if neighbor.enabled == '1' %}
+ neighbor {{ neighbor.address }} remote-as {{ neighbor.remoteas }} 
+{%     endif %}
+{%     if neighbor.enabled == '1' and neighbor.updatesource != '' %}
```

Comment:
If I don't check for neighbor enabled these line will stay if I disable a neighbor. I'll try to put the whole thing under an if as you stated before